url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgList.java

```diff
@@ -0,0 +1,420 @@
+    	}
+    }
+
+    /**
+     * Returns the number of non-option nodes at the base level of this {@code
+     * ArgList}.
+     * 
+     * @return the number of operands at the base level
+     */
+    public int baseCount () {
+        int baseCount = 0;
+        for (ArgNode node = home; node != null; node = node.getEast()) {
+            if (! node.isOption()) {
+                ++baseCount;
+            }
+        }
+        return baseCount;
+    }
+    
+    /**
+     * Transfers items to this {@code ArgList} from another {@code ArgList}.
+     * The home node of the from list replaces the current node of this list.
+     * If the current node was optional, the items added in the southern direction
+     * will also be made optional.
+     * The current node is set to the last node in the eastern direction.
+     * 
+     * @param from the other {@code ArgList}
+     */
+    public void transfer (ArgList from) {
+        boolean optional = curr.isOptional();
+        ArgNode node = from.home;
+        curr.setEast(node.getEast());
+        curr.setSouth(node.getSouth());
+        curr.setEntry(node.getEntry());
+        curr.setRepeat(node.isRepeat());
+        if (optional && (curr.getSouth() != null)) {
+            curr.setSpec(1);
+        }
+        ArgNode last = null;
+        while ((node = node.getSouth()) != null) {
+            last = node;
+            node.setOptional(optional);
+            if (node.getSpec() == 0) {
+                node.setSpec(3);
+            }
+        }
+        if (last != null) {
+            last.setSpec(2);
+        }
+        endEast();
+    }
+    
+    /**
+     * Returns the usage text formatted according to this {@code ArgList}.
+     * 
+     * @return the usage text
+     */
+    public String buildUsage () {
+        StringBuilder sb = new StringBuilder(256);
+        sb.append("Usage:\n");
+        String prog = home.getOperand().getName();
+        int indent = prog.length() + 3;
+        for (ArgNode base = home; base != null; base = base.getSouth()) {
+            int slen = sb.length();
+            sb.append("  ");
+            sb.append(prog);
+            sb.append(' ');
+            buildList(sb, slen, base.getEast(), indent);
+            sb.append("\n");
+        }
+        return sb.toString();
+    }
+    
+    /**
+     * Returns a particular usage specification of the complete usage text.
+     * 
+     * @param base the base {@code ArgNode} for the usage specification
+     * @return the usage specification string
+     */
+    public String buildUsageSpec (ArgNode base) {
+        StringBuilder sb = new StringBuilder(128);
+        String prog = base.getOperand().getName();
+        int indent = prog.length() + 1;
+        int slen = 0;
+        sb.append(prog);
+        sb.append(' ');
+        buildList(sb, slen, base.getEast(), indent);
+        return sb.toString();
+    }
+    
+    private int buildList (StringBuilder sb, int slen, ArgNode node, int indent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.